url/url.pyx

```diff
@@ -0,0 +1,265 @@
+        if encoding == 'utf-8':
+            return cls(s)
+        else:
+            return cls(s.decode(encoding).encode('utf-8'))
+    else:
+        return cls(s.encode('utf-8'))
+
+
+cdef class URL:
+    '''Wrapper around url-cpp.'''
+
+    cdef Url *ptr
+
+    parse = classmethod(ParseMethod)
+
+    def __cinit__(self, const string& s):
+        self.ptr = new Url(s)
+
+    def __dealloc__(self):
+        del self.ptr
+
+    property scheme:
+        def __get__(self):
+            return self.ptr.scheme()
+        def __set__(self, s):
+            self.ptr.setScheme(s)
+
+    property host:
+        def __get__(self):
+            return self.ptr.host()
+        def __set__(self, s):
+            self.ptr.setHost(s)
+
+    property port:
+        def __get__(self):
+            return self.ptr.port()
+        def __set__(self, i):
+            self.ptr.setPort(i)
+
+    property path:
+        def __get__(self):
+            return self.ptr.path()
+        def __set__(self, s):
+            self.ptr.setPath(s)
+
+    property params:
+        def __get__(self):
+            return self.ptr.params()
+        def __set__(self, s):
+            self.ptr.setParams(s)
+
+    property query:
+        def __get__(self):
+            return self.ptr.query()
+        def __set__(self, s):
+            self.ptr.setQuery(s)
+
+    property fragment:
+        def __get__(self):
+            return self.ptr.fragment()
+        def __set__(self, s):
+            self.ptr.setFragment(s)
+
+    property userinfo:
+        def __get__(self):
+            return self.ptr.userinfo()
+        def __set__(self, s):
+            self.ptr.setUserinfo(s)
+
+    def copy(self):
+        '''Return a new instance of an identical URL.'''
+        new = URL('')
+        new.ptr.setScheme(self.ptr.scheme())
+        new.ptr.setHost(self.ptr.host())
+        new.ptr.setPort(self.ptr.port())
+        new.ptr.setPath(self.ptr.path())
+        new.ptr.setParams(self.ptr.params())
+        new.ptr.setQuery(self.ptr.query())
+        new.ptr.setFragment(self.ptr.fragment())
+        new.ptr.setUserinfo(self.ptr.userinfo())
+        return new
+
+    def equiv(self, other, encoding='utf-8'):
+        '''Return true if this url is equivalent to another'''
+        if isinstance(other, basestring):
+            _other = self.parse(other, encoding)
+        else:
+            _other = self.parse(other.utf8)
+
+        _self = self.parse(self.utf8, 'utf-8')
+        _self.canonical().defrag().abspath().escape().punycode().remove_default_port()
+        _other.canonical().defrag().abspath().escape().punycode().remove_default_port()
+
+        return (
+            _self.scheme    == _other.scheme    and
+            _self.host      == _other.host      and
+            _self.path      == _other.path      and
+            _self.params    == _other.params    and
+            _self.query     == _other.query     and
+            _self.port      == _other.port
```

Comment:
Why is `port` last in this check rather than after the `host` to match order of appearance?